### ISSUE TITLE
[Question/Suggestion] Adding Features Without Breaking Configs?

### ISSUE THREAD

**Issue by joshcancode** [danielkrupinski/Osiris#190]
Sorry if this has been asked before, but I was wondering if I can add features without breaking configs? Like I'm trying to add **master switches** for aimbot and visuals, but whenever I add them it breaks my previous configs. Any assistance would be massively appreciated!

Maybe add master switches for aim and visuals in a future update natively? Thank you!

**Comment by memeToasty** [comment]
I tried to add a version control to prevent old configs to be applied to the newer version, but this doesnt work atm. I think you have to rewrite the Config save routine, so that it stores the config in a different format other than binary (i guess, not sure how `ArchiveX.h` works :P)

**Comment by HighPolyLuna** [comment]
> I tried to add a version control to prevent old configs to be applied to the newer version, but this doesnt work atm. I think you have to rewrite the Config save routine, so that it stores the config in a different format other than binary (i guess, not sure how `ArchiveX.h` works :P)

Yeah, I was thinking of coding an XML saving system, but I knew it'd be a lot of work and thought I'd ask here. Let me know if your version control takes off!

**Comment by memeToasty** [comment]
Yeah my current problem is, that the Config is working with `struct` Structure, and its not so easy to convert these to an array or a dictionary to save the config as clear text... Btw. I've never worked with ´struct´s, so dont expect my version control too early...

**Comment by memeToasty** [comment]
I've looked into the `Config.cpp` and the `Config.h` again, to find a way of saving the configs so, that they will load, even if a new feature will be implemented.. Unfortunately this seems not possible at the moment, as the config is using constexpr functions to store the individual weapon settings. So you cannot easily write the write the config to a json file or whatever without rewriting the whole cheat to use another way of storing configs. Also I am not that experienced at that level of C++ to easily rewrite the whole software... It would be a possibility to manually format ALL the config variables to a readable format, but that would mean that every new setting has to be implemented in this function manually.. Also as the weapon settings are being worked with a function you cannot simply write out all the config settings into one simple file... I think at the moment it is the best for all to just rework your configs every time an update occurs... I already implemented a method, that stops older configs to be loaded in, thus loading in the settings wrongly, on my fork.. I also entered a pull request with that feature... Maybe Daniel confirms this feature and in the future we dont have to face this kind of loading issue.
